diffkemp/simpll/SMTBlockComparator.cpp

```diff
@@ -16,7 +16,67 @@
 
 using namespace llvm;
 
+void SMTBlockComparator::findSnippetEnd(BasicBlock::const_iterator &InstL,
+                                        BasicBlock::const_iterator &InstR) {
```

Comment:
Note: this is something we will need to other extensions, too (e.g. equivalence slicing introduced by #182) so we should at least think about a better place for this method b/c we'll want to have it unified.

Reply:
Idea: we could have an abstract class (naming TBD, e.g. `SearchSynchronizationComparator` comes to mind) that has a single field (of type `DifferentialFunctionComparator`) and implements this method. The slicer and SMTBlockComparator could then inherit from this class

Reply:
Yes, that sounds good. An alternative would be to have a helper class which does pretty much the same but the slicer and `SMTBlockComparator` use it as a field. I'm not sure which approach would be better so I'd stick with the current implementation for now and introduce the abstraction only when necessary.